### ISSUE TITLE
Age of flow records

### ISSUE THREAD

**Issue by sjoki** [aperezdc/snabbswitch#17]
Hi,
I haven't seen any parameters which limit for the period of state records life. I haven't found similar parameters in the ndpi code. But my analysis was not so serious. Do I understand correctly that I need to clean up old state records myself? I'm worried about using memory.

**Comment by aperezdc** [comment]
@sjoki: That's correct, freeing old flows is not done automatically, in order to allow for more flexibility. Each application may have different needs in this regard, so it seems like a better fit to let them choose what to do. That being said, probably we should provide some API in the `Scanner` to dispose of flows. 

I think we could add the following method:

```lua
function Scanner:eject_flow(flow)
  -- Asks the Scanner to stop tracking the given "flow", possibly freeing
  -- resources. The "flow" object is still valid after this method returns
  -- but its fields will NOT be updated anymore.
  --
  -- This method cannot be used while iterating over the detected flows with
  -- Scanner:flows(). Instead, gather the flows to be ejected in a list, and
  -- use this method on them after iterating has finished.
end
```

The rationale behind the first comment is that flows mostly will be tables containing data fields, or if they are a C-based userdata or FFI C data, they should have a `__gc` metamethod responsible of freeing them. Ejecting a flow may free auxiliar resources that the `Scanner` may be using to track the flow.

As per the comment about iteration, the idea is that invalidating iterators can get tricky and also be additionally cumbersome when mixing C and Lua code. Therefore it seems like a reasonable compromise to not do it at all, and instead require that the client code gathers a list of elements to remove while iterating, and removing afterwards. For example, this approach is already needed when using Lua's `pairs()` so it shouldn't come as a surprise :wink: 

@takikawa: Any thoughts on this?
